### PR TITLE
Adding a custom background color property

### PR DESCRIPTION
**Pull request by fabricciotc**
We are adding a new `bgColor` property to set a specific background color to our notification containers
Currently, we just can use any of these custom colors: success, info, warning, danger, advisor

### CODE REVIEW

File: packages/es-components/src/components/containers/notification/useNotification.js [hunk 1]
```diff
@@ -65,7 +71,7 @@ function NotificationContent(props) {
 
 const Notification = styled.div`
   align-items: center;
-  background-color: ${props => props.variant.bgColor};
+  background-color: ${props => props.bgColor ?? props.variant.bgColor};
```
Comment (fabricciotc) [inline packages/es-components/src/components/containers/notification/useNotification.js:74]: If `props.bgColor` is null, the background color will use from `props.variant.bgColor`

File: packages/es-components/src/components/containers/notification/useNotification.js [hunk 2]
```diff
@@ -125,6 +132,7 @@ export function useNotification(styleType = 'base') {
           variant={color}
           className={className}
           style={style}
+          bgColor={bgColor}
```
Comment (fabricciotc) [inline packages/es-components/src/components/containers/notification/useNotification.js:135]: This is the base component for LightNotification and Notification. We will add this property to set a specific background color

**Review by Darrken — APPROVED** [review]

**Comment by bradyclifford** [comment]
Would be helpful if you include an example using that new prop in the docs.

File: packages/es-components/src/components/containers/notification/useNotification.js (hunk 1 above)
Comment (stevematney) [reply packages/es-components/src/components/containers/notification/useNotification.js:74]: I'm not sure this work is actually necessary, and I think it convolutes the API.

`Notification` is required to be used with a named `type` prop, which is responsible for all the styles in that Notification to correlate with our design system.

If you need a custom Notification, you can always extend it with styled components like this:
```js
const SpecialNotification = styled(Notification)`
  background-color: ${props => props.bgColor};
}`;

// ...later
function SpecialDisplayNotification {
  return (
    <SpecialNotification bgColor="perwinkle">
      This notification is very special!
    </SpecialNotification>
  );
}
```

**Review by stevematney — CHANGES_REQUESTED** [review]

**Comment by bradyclifford** [comment]
I agree with Steve that we should keep the `type` prop responsible for the styles exposed by our design system.  One-off changes that are not specific to our design system can be done as Steve suggested using a wrapped style-component.

With that said, I would argue that if we see in our latest UX wireframe designs (amongst all the teams) that we are continuously adding custom styled-components to adjust the default background color, perhaps we are missing a type?  `success, info, warning, danger, advisor`

**Comment by stevematney** [comment]
> With that said, I would argue that if we see in our latest UX wireframe designs (amongst all the teams) that we are continuously adding custom styled-components to adjust the default background color, perhaps we are missing a type? `success, info, warning, danger, advisor`

Right. If we are wholly missing a type of Notification ("Alert" in the design system docs), or if we need an additional one, we can add a new one. I would defer to Keoni and @wtw-austinandriese to answer that question for us, since this is really a decision about the design system.

**Comment by bradyclifford** [comment]
If this deviation is very much unique to the IFP team, it either needs to remain a deviation and not added to ES Components (as the exception not the norm) or the team needs to consider coming back to the standard and use one of the existing types.

**Pull request closed without merging**